script/openqa-clone-job

```diff
@@ -207,6 +211,8 @@ sub parse_options () {
     die "missing job reference, see --help for usage\n" unless $jobid;
     ($options{'from'}, $jobid) = split_jobid($jobid) unless $options{'from'};
     usage(1) unless ($jobid && $options{'from'});
+    $options{'repeat'} ||= 1;
+    die "invalid repeat count, should be between 1 and 999" if ($options{'repeat'} < 1 or $options{'repeat'} > 999);
```

Comment:
This should probably check whether the user-specified value is actually a number:
```suggestion
    my $repeat = $options{'repeat'} //= 1;
    die "invalid repeat count, should be between 1 and 999\n" if !looks_like_number($repeat) || $repeat < 1 || $repeat > 999;
```

I would also prefer `||` over `or` except for command-chaining. Adding an `"\n"` at the end of the error message prevents printing the line number making this user error more look like one.

You also have to use `//=` instead of `||=` because otherwise the error message would not be shown in case the user specifies `0`.

And as others have already suggested it would make sense to remove the rather arbitrary limit and use @perlpunk 's suggestion to handle it.

Reply:
good point, thanks for the suggestion. I will change the code and adjust unit test accordingly.